cws-engine-service/src/main/java/jpl/cws/engine/listener/ProcessEventListener.java

```diff
@@ -81,6 +81,7 @@ else if (eventType.equals("sync")) {
 								
 								// If the process counter state changed, then we potentially have more bandwidth to
 								// execute more processes of the type that was just completed/failed.
+								// QUESTION: whats the reason to have this? It would take away bandwith for other procs
```

Comment:
This line may need to be reworked now, with the new scheme.  In the past, there was only the limits per procDefKey, and if one procDefKey finished, then you would want to see if you could immediately start running another one (of the same procDefKey type).  Now that the overall max is potentially across multiple procDefKeys, then I think this line may cause issues with the new scheme. I think instead calling the claimHighestPriorityStartReq chain of actions, so that it goes and looks for more work through the new way of doing things, would be the best way here.

Reply:
To get to `claimHighestPriorityStartReq` and claim a set of multiple procDefKeys, it would need the List of enabled Proc Definitions.   

Since the Max Running Limit now covers all procs, wouldn’t it be fine to remove the `limitToProcDefKey` condition thats inside the WorkerService.claimWithCounter ?  _L687-L689_

https://github.com/NASA-AMMOS/common-workflow-service/blob/06122f3f561188d350ceed834a9c0c631beb437f/cws-engine-service/src/main/java/jpl/cws/engine/WorkerService.java#L684-L690

This way the ProcessEventListener still has the message for the Process EndEvent and returns quickly to the procStartReqAction to claim more, BUT it claims the prioritized process from the bunch instead of the Single procDefKey that was passed.

Reply:
Another, change I made is switching the start req action from `procDefKey` to `null` . This way when the process ends, it sends a message and returns to WorkerService to pick up more claims but doesn't restrict to a single procDefKey.

https://github.com/NASA-AMMOS/common-workflow-service/blob/cc395ba1eb3950607186a2faebecf01d504b3d7e/cws-engine-service/src/main/java/jpl/cws/engine/listener/ProcessEventListener.java#L81


Similar to what happens in the WorkerDaemon here:

https://github.com/NASA-AMMOS/common-workflow-service/blob/cc395ba1eb3950607186a2faebecf01d504b3d7e/cws-engine-service/src/main/java/jpl/cws/engine/WorkerDaemon.java#L87